### PR TITLE
fix: events in bridgeless mode

### PR DESCRIPTION
**Pull request by WoLewicki**
## PR concerning New Architecture support in the library :tada:

We at [Software Mansion](https://swmansion.com/) have been working on [improving support](https://blog.swmansion.com/sunrising-new-architecture-in-the-new-expensify-app-729d237a02f5) for the new architecture for quite a while now. If you need help with anything related to New Architecture, like:
- [migrating your library](https://x.com/swmansion/status/1717512089323864275)
- [migrating your app](https://github.com/Expensify/App/pull/13767)
- [investigating issues](https://github.com/facebook/react-native/pulls?q=sort%3Aupdated-desc+is%3Apr+author%3Aj-piasecki+is%3Aopen)
- [improving performance](https://x.com/BBloniarz_/status/1808138585528303977)

or you just want to ask any questions, hit us up on [projects@swmansion.com](mailto:projects@swmansion.com)

---

## What

PR fixing events in bridgeless mode on Android, due to their stricter convention of naming there.

### CODE REVIEW

File: package/android/src/main/java/com/mrousavy/camera/react/Events.kt [hunk 1]
```diff
@@ -5,64 +5,100 @@ import com.facebook.react.bridge.WritableMap
 import com.facebook.react.uimanager.events.Event
 
 class CameraInitializedEvent(surfaceId: Int, viewId: Int) : Event<CameraInitializedEvent>(surfaceId, viewId) {
-  override fun getEventName() = "cameraInitialized"
+  override fun getEventName() = EVENT_NAME
   override fun getEventData(): WritableMap = Arguments.createMap()
+  companion object {
+    const val EVENT_NAME = "topInitialized"
```
Comment (mrousavy) [inline package/android/src/main/java/com/mrousavy/camera/react/Events.kt]: Why top instead of camera?

**Review by hannojg — APPROVED** [review]
lgtm, thanks!

**Review by mrousavy — COMMENTED** [review]
Hey - thanks for this PR, really helpful stuff! 🙏
Just one question

File: package/android/src/main/java/com/mrousavy/camera/react/Events.kt (hunk 1 above)
Comment (hannojg) [reply package/android/src/main/java/com/mrousavy/camera/react/Events.kt]: every event needs to be prefixed with `top` now in the event emitter code. @WoLewicki Can we please rename these events to `topCameraInitialized` etc

**Review by hannojg — CHANGES_REQUESTED** [review]
@WoLewicki Can we please rename these events to `topCameraInitialized` (so including Camera) etc?

**Comment by hannojg** [comment]
All good, already went ahead and updated the names. @mrousavy good for merge!

**Comment by WoLewicki** [comment]
@hannojg are you sure it works? I remember having some problems with such renaming, I think it is due to JS callbacks not having `camera` in their name. Maybe wait with merging it and let me double-check ok?

**Comment by hannojg** [comment]
I tested it with the renaming and it was working in the example app, so should be good 😊  If you have the time feel free to double check ofc 🫶

**Review by hannojg — APPROVED** [review]

File: package/android/src/main/java/com/mrousavy/camera/react/CameraViewManager.kt [hunk 2]
```diff
@@ -29,18 +29,18 @@ class CameraViewManager : ViewGroupManager<CameraView>() {
...
+      .put(CameraStartedEvent.EVENT_NAME, MapBuilder.of("registrationName", "onStarted"))
+      .put(CameraStoppedEvent.EVENT_NAME, MapBuilder.of("registrationName", "onStopped"))
+      .put(CameraShutterEvent.EVENT_NAME, MapBuilder.of("registrationName", "onShutter"))
+      .put(CameraErrorEvent.EVENT_NAME, MapBuilder.of("registrationName", "onError"))
+      .put(CameraCodeScannedEvent.EVENT_NAME, MapBuilder.of("registrationName", "onCodeScanned"))
+      .put(CameraPreviewStartedEvent.EVENT_NAME, MapBuilder.of("registrationName", "onCodeScanned"))
+      .put(CameraPreviewStoppedEvent.EVENT_NAME, MapBuilder.of("registrationName", "onCodeScanned"))
+      .put(CameraOutputOrientationChangedEvent.EVENT_NAME, MapBuilder.of("registrationName", "onCodeScanned"))
+      .put(CameraPreviewOrientationChangedEvent.EVENT_NAME, MapBuilder.of("registrationName", "onCodeScanned"))
+      .put(AverageFpsChangedEvent.EVENT_NAME, MapBuilder.of("registrationName", "onCodeScanned"))
```
Comment (mrousavy) [inline package/android/src/main/java/com/mrousavy/camera/react/CameraViewManager.kt]: These 5 events are wrong. Please correct the names

File: package/android/src/main/java/com/mrousavy/camera/react/Events.kt [hunk 3]
```diff
@@ -5,64 +5,100 @@ import com.facebook.react.bridge.WritableMap
...
+    const val EVENT_NAME = "topCameraPreviewOrientationChanged"
+  }
 }
 
 class AverageFpsChangedEvent(surfaceId: Int, viewId: Int, private val data: WritableMap) : Event<CameraShutterEvent>(surfaceId, viewId) {
-  override fun getEventName() = "averageFpsChanged"
+  override fun getEventName() = EVENT_NAME
   override fun getEventData() = data
+  companion object {
+    const val EVENT_NAME = "topAverageFpsChanged"
```
Comment (mrousavy) [inline package/android/src/main/java/com/mrousavy/camera/react/Events.kt]:
```suggestion
    const val EVENT_NAME = "topCameraAverageFpsChanged"
```

**Review by mrousavy — CHANGES_REQUESTED** [review]

File: package/android/src/main/java/com/mrousavy/camera/react/CameraViewManager.kt (hunk 2 above)
Comment (mrousavy) [inline package/android/src/main/java/com/mrousavy/camera/react/CameraViewManager.kt]:
```suggestion
      .put(CameraPreviewStartedEvent.EVENT_NAME, MapBuilder.of("registrationName", "onPreviewStarted"))
      .put(CameraPreviewStoppedEvent.EVENT_NAME, MapBuilder.of("registrationName", "onPreviewStopped"))
      .put(CameraOutputOrientationChangedEvent.EVENT_NAME, MapBuilder.of("registrationName", "onOutputOrientationChanged"))
      .put(CameraPreviewOrientationChangedEvent.EVENT_NAME, MapBuilder.of("registrationName", "onPreviewOrientationChanged"))
      .put(AverageFpsChangedEvent.EVENT_NAME, MapBuilder.of("registrationName", "onAverageFpsChanged"))
```

**Comment by mrousavy** [comment]
I committed the changes to fix the wrong names - @hannojg if you already have the test setup running could you maybe test if everything still works? Especially onPreviewStarted and onPreviewStopped

**Comment by hannojg** [comment]
Sure, let me test, one second

**Comment by hannojg** [comment]
Screenshot 2024-10-16 at 11 16 05

yes, everything is still working on android!

**Review by mrousavy — APPROVED** [review]

**Comment by mrousavy** [comment]
Awesome - LGTM then - thanks @WoLewicki & @hannojg !!! ❤️

**Comment by WoLewicki** [comment]
Hey guys, did you test it in bridgeless mode? It's where the issue exists, as mentioned in the PR title. It's much stricter than bridge mode. cc @hannojg

**Comment by hannojg** [comment]
Will do those tests in a moment, yes! We'll handle everything in a separate PR for new arch 😊 !

**Comment by hannojg** [comment]
Screenshot 2024-10-17 at 16 14 50

Working fine in bridgeless mode!

**Pull request merged**